File: assets/code-variants/ch12/02.py
def test_empty_account_should_not_be_valid():
    assert not processor._is_valid(new_transaction(sender=EMPTY_ACCOUNT))

def test_should_save_serialized_data():
    processor._save_to_database(new_transaction(id=123, sender="me", recipient="you", amount=100))
    assert database.get(123) == "me,you,100"

File: assets/code-variants/ch12/02.java
@Test
public void emptyAccountShouldNotBeValid() {
    assertThat(processor.isValid(newTransaction().setSender(EMPTY_ACCOUNT))).isFalse();
}

@Test
public void shouldSaveSerializedData() {
    processor.saveToDatabase(newTransaction().setId(123).setSender("me").setRecipient("you").setAmount(100));
    assertThat(database.get(123)).isEqualTo("me,you,100");
}
